src/WorldStorage/FastNBT.cpp

```diff
@@ -257,7 +250,7 @@ eNBTParseError cParsedNBT::ReadList(eTagType a_ChildrenType)
 	NEEDBYTES(4, eNBTParseError::npListMissingLength);
 	int Count = GetBEInt(m_Data + m_Pos);
 	m_Pos += 4;
-	if ((Count < 0) || (Count > MAX_LIST_ITEMS))
+	if ((Count < 0) || (Count > static_cast<int>((m_Length - m_Pos) / 4)))
```

Comment:
Why 4? Should this depend on `a_ChildrenType` or is 4 a lower bound?

Reply:
No specific reason. I wanted to keep it simple to read. But you're right, it's bound to fail randomly in the future, I'll rewrite it to take the children size into account.